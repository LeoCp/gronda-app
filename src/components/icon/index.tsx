import React from 'react';
import { Image } from 'react-native';

import { ICONS } from '@components/icon/constants/icons';

type Props = { name: keyof typeof ICONS; height: number; width: number };

export function Icon({ name, height, width }: Props) {
  return <Image source={ICONS[name]} style={{ height, width }} />;
}
