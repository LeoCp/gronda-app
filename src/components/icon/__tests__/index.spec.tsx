import React from 'react';
import { render } from '@testing-library/react-native';

import { Icon } from '@components/icon';

describe('Icon Component', () => {
  it('Should match snapshot', () => {
    const { toJSON } = render(<Icon name="logo" height={10} width={10} />);
    expect(toJSON()).toMatchSnapshot();
  });
});
