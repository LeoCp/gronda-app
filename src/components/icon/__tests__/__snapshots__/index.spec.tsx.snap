// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon Component Should match snapshot 1`] = `
<Image
  source={
    {
      "testUri": "../../../src/assets/icons/logo.png",
    }
  }
  style={
    {
      "height": 10,
      "width": 10,
    }
  }
/>
`;
